pkg/sync/sync_context.go

```diff
@@ -1048,6 +1048,14 @@ func (sc *syncContext) processCreateTasks(state runState, tasks syncTasks, dryRu
 			if !dryRun || sc.dryRun || result == common.ResultCodeSyncFailed {
 				sc.setResourceResult(t, result, operationPhases[result], message)
 			}
+
+			if sc.dryRun {
```

Comment:
Nitpick:

If `sc.dryRun==true` then first `if` at line 1048 should set operation result and then new `if` overrides it. Instead I suggest handle both cases in first if as following:

```golang
if !dryRun || sc.dryRun || result == common.ResultCodeSyncFailed {
	phase := operationPhases[result]
	// no resources are created in dry-run, so running phase means validation was
	// successful and sync operation succeeded
	if sc.dryRun && phase == common.OperationRunning {
		phase = common.OperationSucceeded
	}
	sc.setResourceResult(t, result, phase, message)
}
```

Reply:
Thank you @alexmt !